pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/juniperkits/JuniperKitCreationStatusTest.java

```diff
@@ -86,6 +111,29 @@ public void testStatusKitEndpointByJuniperStudyGuid() {
         }
     }
 
+    @Test
+    public void testCurrentStatusFieldAfterChanges() {
+        int rand = new Random().nextInt() & Integer.MAX_VALUE;
+        JuniperKitRequest juniperTestKit = generateJuniperKit(rand);
+        try {
+            createAndAssertNonPepperCreation(juniperTestKit);
+            KitResponse kitResponse = nonPepperStatusKitService.getKitsBasedOnJuniperKitId(juniperTestKit.getJuniperKitId());
+            assertStatusKitResponse(kitResponse, juniperTestKit, rand, NonPepperStatusKitService.KIT_WITHOUT_LABEL);
+            JuniperSetupUtil.changeKitToQueue(juniperTestKit, mockEasyPostUtil);
+            kitResponse = nonPepperStatusKitService.getKitsBasedOnJuniperKitId(juniperTestKit.getJuniperKitId());
+            assertStatusKitResponse(kitResponse, juniperTestKit, rand, NonPepperStatusKitService.QUEUE);
+            juniperTestKit.setDdpLabel(kitResponse.getKits().get(0).getDsmShippingLabel());
+            JuniperSetupUtil.changeKitToSent(juniperTestKit);
+            kitResponse = nonPepperStatusKitService.getKitsBasedOnJuniperKitId(juniperTestKit.getJuniperKitId());
+            assertStatusKitResponse(kitResponse, juniperTestKit, rand, NonPepperStatusKitService.SENT);
+            JuniperSetupUtil.changeKitToReceived();
+            kitResponse = nonPepperStatusKitService.getKitsBasedOnJuniperKitId(juniperTestKit.getJuniperKitId());
+            assertStatusKitResponse(kitResponse, juniperTestKit, rand, NonPepperStatusKitService.RECEIVED);
+        } finally {
+            createdKitIds.add(juniperTestKit.getJuniperKitId());
```

Comment:
Is there a reason to have this in a finally block? It seems like it could go immediately after the call to `generateJuniperKit`.

Reply:
I want to make sure if anything goes wrong while the new kit is getting added the new kit still gets added to the createdKitIds so it get deleted in the end

Reply:
Right. But isn't that accomplished by just adding the kit ID right after the call to `generateJuniperKit`? Just from the structure of the code (and not knowing the content) that seems true.

Reply:
Right, I removed ` createdKitIds.add(juniperTestKit.getJuniperKitId());` from finally and moved it into `createAndAssertNonPepperCreation`